src/hotspot/make/bsd/makefiles/gcc.make

```diff
@@ -363,7 +373,15 @@ ASFLAGS += -x assembler-with-cpp
 # Linker flags
 
 # statically link libstdc++.so, work with gcc but ignored by g++
-STATIC_STDCXX = -Wl,-Bstatic -lstdc++ -Wl,-Bdynamic
```

Comment:
I don't think you have to check OS_VENODR  here.  Basically, it's clang/gcc thing. 
g++ uses libstdc++. clang++ links libc++.  It should stick with libc++ even on linux platform.